### ISSUE TITLE
Int-8 converter scales static model parameters incorrectly

### ISSUE THREAD

**Issue by JvKgithub** [tensorflow/tensorflow#56739]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installation (pip package or built from source): pip
- TensorFlow library (version, if pip package or github SHA, if built from source): 2.9.1 (tensorflow-estimator 2.9.0)

### 2. Code

Provide code to help us reproduce your issues using one of the following options:

https://colab.research.google.com/drive/1HN3tZCEeK79YPFT85udXf7FybBo6-PYf

### 3. Failure after conversion
If the conversion is successful, but the generated model is wrong, then state what is wrong:

After INT-8 conversion, the models parameters are scaled incorrectly (and unnecessarily) and model performance drops significantly.
The colab notebook compares a default conversion and a 8-bit quantized conversion and shows that the output changes and difference between classes can no longer be made. The same code produces the same issues with the upper mentioned TF installation and in the default colab notebook configuration.

The model consists of a simple custom layer that compares the input bounding box with an internal hardcoded bounding box. Only minimum, maximum and subtract operations are used. 
When converting the Keras sequential model to a default .tflite model (not quantized), the output is exactly as expected. Converting it with 8-bit quantization the model parameters are scaled in a way that clear, non-borderline cases are predicted incorrectly. 

I understand that scaling might be necessary for large models with diverse inputs, but in this case I would like have full control over what changes are made to the model during conversion, since it is a very simple model and the possible input is very predictable and narrowly defined.
Inputs and model parameters both range from 0-100, for which no scaling is needed and no overflow could possibly happen. This is (I hope) reflected in my representative dataset, which seems to be a necessary input for 8-bit integer quantization.

I do not mind the exact numerical output of the model, but the output of the model should be different for overlapping and non-overlapping bounding boxes, which is only given for the default converted tflite model.

To sum up, how can I convert or write a model for 8bit integers, in a way that I have direct control over the model parameters.

**Comment by mohantym** [comment]
Hi @gowthamkpr !
Could you please look at this issue. Attached gist in [2.8](https://colab.sandbox.google.com/gist/mohantym/2bc50db5d98fdb293371bb7c5cacb3e1/convert_int8.ipynb#scrollTo=9kzTojLK_t74), [2.9](https://colab.sandbox.google.com/gist/mohantym/d6a0e1f3778b27f5a9d95f87550aad47/convert_int8.ipynb#scrollTo=9kzTojLK_t74) and [nightly](https://colab.sandbox.google.com/gist/mohantym/3c627ae916969867de9ec0a92d035912/convert_int8.ipynb#scrollTo=9kzTojLK_t74) for reference.
Thank you!

**Comment by JvKgithub** [comment]
Are there any news on this issue @miaout17 ?

**Comment by pjpratik** [comment]
I was able to reproduce this issue in TF 2.11 and TF Nightly 2.12.0-dev20230111. Please find the gist [here](https://colab.sandbox.google.com/gist/pjpratik/448a9c6840a53b8dff21d4b20ede833f/56739.ipynb#scrollTo=KX0BqNjxuhV-). Thank you.

**Comment by pkgoogle** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFLite team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the TFLite space.
